src/commands/registry.rs

```diff
@@ -86,7 +86,14 @@ impl Push {
 
         let _spinner = create_dotted_spinner(2000, "Pushing app to the Registry".to_owned());
 
-        let digest = client.push(&app_file, &self.reference, annotations).await?;
+        let digest = client
+            .push(
+                &app_file,
+                &self.reference,
+                annotations,
+                PredefinedAnnotations::Infer,
```

Comment:
So the behavior here is: always infer, and if the user provided any annotation explicitly, it takes priority over the inferred value??

Reply:
If the user provides _one of the annotations that would be inferred_ explicitly, then the explicit value takes precedence - see the comment on the enum.